### PR TITLE
Vue: add login context with simple form page

### PR DESCRIPTION
**Pull request by matthieuRioual**
- related to #897

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueDomainService.java [hunk 1]
```diff
@@ -210,4 +210,15 @@ public void addJestSonar(Project project) {
...
+  public void addLoginContext(Project project) {
+    String destinationPrimaryLoginContext = "src/main/webapp/app/common/primary/login";
+    String sourcePrimaryLoginContext = "webapp/app/jwt/primary/login";
+    projectRepository.template(project, getPath(SOURCE, "webapp/app/jwt/domain"), "Login.ts", "src/main/webapp/app/common/domain");
+    projectRepository.template(project, getPath(SOURCE, sourcePrimaryLoginContext), "index.ts", destinationPrimaryLoginContext);
+    projectRepository.template(project, getPath(SOURCE, sourcePrimaryLoginContext), "Login.component", destinationPrimaryLoginContext);
+    projectRepository.template(project, getPath(SOURCE, sourcePrimaryLoginContext), "Login.html", destinationPrimaryLoginContext);
+    projectRepository.template(project, getPath(SOURCE, sourcePrimaryLoginContext), "Login.vue", destinationPrimaryLoginContext);
+    projectRepository.template(project, getPath(SOURCE, "webapp/app/jwt/secondary"), "LoginDTO.ts", "src/main/webapp/app/common/secondary");
+  }
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueDomainService.java:223]: as discussed, this part should not be included in `client/vue/core`, it should be in a separated context like `client/vue/security/jwt`, so later we can have `client/vue/security/oauth2`

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
I'm converting this PR to draft, as it's not ready yet

**Comment by pascalgrimaud** [comment]
Can you have a look plz @seraphinandrieux ?
I'll do the same when possible

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwt.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import tech.jhipster.lite.common.domain.FileUtils;
+import tech.jhipster.lite.generator.project.domain.Project;
+
+public class VueJwt {
+
+  private VueJwt() {}
+
+  public static final String PACKAGE_JSON = "package.json";
+
+  public static boolean checkIfPinia(Project project) {
```
Comment (seraphinandrieux) [inline src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwt.java:15]:
```suggestion
  public static boolean isPiniaImplemented(Project project) {
```

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import tech.jhipster.lite.error.domain.Assert;
+import tech.jhipster.lite.error.domain.GeneratorException;
+import tech.jhipster.lite.generator.project.domain.Project;
+import tech.jhipster.lite.generator.project.domain.ProjectRepository;
+
+public class VueJwtDomainService implements VueJwtService {
+
+  public static final String SOURCE = "client/vue";
+
+  public static final String DESTINATION_PRIMARY = "src/main/webapp/app/common/primary/";
```
Comment (seraphinandrieux) [inline src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java:14]: I believe you have these const already set into a class. @pascalgrimaud I cannot remember which one ?

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  private final ProjectRepository projectRepository;
+
+  public VueJwtDomainService(ProjectRepository projectRepository) {
+    this.projectRepository = projectRepository;
+  }
+
+  @Override
+  public void addJWT(Project project) {
+    Assert.notNull("project", project);
+    if (!VueJwt.checkIfPinia(project)) throw new GeneratorException("Pinia has not been added");
```
Comment (seraphinandrieux) [inline src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java:32]: If you use only once this method, you should problably update it into `isPiniaNotImplemented`

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  private final ProjectRepository projectRepository;
+
+  public VueJwtDomainService(ProjectRepository projectRepository) {
+    this.projectRepository = projectRepository;
+  }
+
+  @Override
+  public void addJWT(Project project) {
+    Assert.notNull("project", project);
```
Comment (seraphinandrieux) [inline src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java:31]: Besides extract the two assertions

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    if (!VueJwt.checkIfPinia(project)) throw new GeneratorException("Pinia has not been added");
+    addLoginContext(project);
+  }
+
+  public void addLoginContext(Project project) {
+    String destinationPrimaryLoginContext = DESTINATION_PRIMARY + "/login";
+    String sourcePrimaryLoginContext = SOURCE_PRIMARY + "/login";
+    projectRepository.template(project, getPath(SOURCE, SOURCE_DOMAIN), "Login.ts", DESTINATION_DOMAIN);
```
Comment (seraphinandrieux) [inline src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java:39]: Extract all files which have to be generated into Domain, and do a for each here to call template method

File: src/test/java/tech/jhipster/lite/generator/client/vue/security/jwt/infrastructure/primary/rest/VueJwtResourceIT.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.test.web.servlet.MockMvc;
+import tech.jhipster.lite.IntegrationTest;
+import tech.jhipster.lite.generator.init.application.InitApplicationService;
+
+@IntegrationTest
+@AutoConfigureMockMvc
+class VueJwtResourceIT {
```
Comment (seraphinandrieux) [inline src/test/java/tech/jhipster/lite/generator/client/vue/security/jwt/infrastructure/primary/rest/VueJwtResourceIT.java:11]: I think it misses some tests

**Review by seraphinandrieux — APPROVED** [review]

File: src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java (hunk 3 above)
Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/client/vue/security/jwt/domain/VueJwtDomainService.java:14]: Yes, simply see Constants

**Comment by pascalgrimaud** [comment]
@matthieuRioual : can it be closed in favor of https://github.com/jhipster/jhipster-lite/pull/1759 ?

**Pull request closed without merging**